PROVESFlightControllerReference/test/unit-tests/test_PacketProcessor_Validator.cpp

```
@@ -0,0 +1,87 @@
+    Packet p{0u, 20u, 0x01000000u, {}};  // sequence 20, expected 10, window 5 but bypass opcode
+    auto res = validatePacket(p, 10u, 5u);
+    EXPECT_EQ(res, PacketValidator::Status::Bypass);
+}
+
+TEST(PacketValidatorTest, SpiInvalid) {
+    Packet p{1u, 10u, 0xDEADBEEFu, {}};  // non-zero SPI invalid
+    auto res = validatePacket(p, 10u, 5u);
+    EXPECT_EQ(res, PacketValidator::Status::SpiInvalid);
+}
+
+TEST(PacketValidatorTest, SequenceNumberInvalid) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)